src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -600,14 +599,19 @@ def set_names(
         >>> idx.set_names('quarter')
         Index([1, 2, 3, 4], dtype='int64', name='quarter')
         """
-        # TODO: SNOW-1458122 implement set_names
-        WarningMessage.index_to_pandas_warning("set_names")
-        if not inplace:
-            return Index(
-                self.to_pandas().set_names(names, level=level, inplace=inplace),
-                convert_to_lazy=self.is_lazy,
-            )
-        return self.to_pandas().set_names(names, level=level, inplace=inplace)
+        if not isinstance(names, list):
+            names = [names]
+
+        if self.is_lazy:
+            if not inplace:
+                return Index(
+                    self._query_compiler.set_index_names(names),
+                )
+            else:
+                self._set_names(names)
+                return None
+        else:
+            return self.to_pandas().set_names(names, level=level, inplace=inplace)
```

Comment:
`_set_names` has logic to handle the non-lazy version of setting values without calling `to_pandas` is it possible to reuse that functionality or does that handle a different case?

Reply:
That handles a different case, that has no return, so it can be reused for the inplace method. However, when we are not in place, we need to create a new index, so we use the query compiler method `set_index_names`, which creates a new query compiler to pass to the index constructor